Replace build tool maven to sbt

takezoux2 opened the pull request:
- Replace scala versions to property in pom.xml
- Append scala compat version to artifactId(It's a standard name rule for scala libraries)
- Upgrade scala version to 2.11.8
- Add build.sbt to also build by sbt
- Remove 2.10.4 build in travis. Because scala 2.10.x build has already broken because of scala-logging.(Scala logging does not support scala 2.10.X)

Scala 2.12 coming soon.So should support multi scala versions.

And I have a question. .gitignore file is removed. What was wrong?

takezoux2 committed:
Append scala compat version.
Support sbt build.

takezoux2 committed:
Fix travis scala version

takezoux2 committed:
Make skip to test in travis build.

takezoux2 committed:
Move macro codes to each scala version directory.
Remove 2.10 support because scala-logging does not support scala 2.10.X.

takezoux2 committed:
Merge branch 'master' into improve_build

# Conflicts:
#	README.md
#	pom.xml

takezoux2 committed:
Fix library dependency.

takezoux2 committed:
Fix README.md

nikita-volkov commented:
If you want to switch to SBT, you can just make the switch. Doubling of the project build files will prove to be a maintenance pain-point. Otherwise let it be Maven. All I'm asking is just to pick one.

> And I have a question. .gitignore file is removed. What was wrong?

It's the developer's private thing, which environment he has, and hence which files to ignore. You can have your version of `.gitignore` on your machine and I'll have a different one on mine. The changes to it have nothing to do with the project.

To have `.gitignore` ignored by Git just mention it in itself or add it to your global list.

takezoux2 committed:
Update .travis.yml

takezoux2 committed:
Disable parallel execution in test

takezoux2 committed:
Fix to pass test.
### ReflectionSuite.scala
 fix reflection name.
### FieldTestSuite1.scala
 Comment out oracle test. This test was skipped in maven test.

nikita-volkov commented:
> Sharing .gitignore prevents miss adding.

It does not. It only prevents miss-adding of files, which are mentioned in it. The ones that aren't can still be added. To make your statement true, all the possible names, which don't relate to the project need to be mentioned. 

A person is responsible for what he does, a committer is expected to review his commits.

I'm pretty sure you wouldn't be satisfied with [one of the "gitignore" files that you've mentioned](https://github.com/play/play/blob/master/.gitignore). So you'd be doing commits to that file. So, merges, discussions... All about what? Something that has zero value for the project.

So I've found it wisest to just keep out of this problem and hence keep it out of my projects. Trust me, once you'll add the ".gitignore" line to your local ".gitignore" file you'll forget we ever had this conversation.

> SBT have a great advantage to cross build.

No problem. Switch to it, if you feel the need to.

> OK, l'll soon write sbt's pom generator.

What?

takezoux2 committed:
remove .gitignore

takezoux2 committed:
Remove pom.xml
Make to generate pom.xml by "sbt makePom" command

takezoux2 commented:
Remove .gitignore and pom.xml.
But you can generate pom.xml by "sbt makePom".

takezoux2 committed:
delete pom.xml

pjfanning reviewed (changes requested):
Could you change the build.sbt common settings sorm version to 0.3.22-SNAPSHOT?
Also, could you change the sext version to 0.2.26.
I would also like to see the c3p0 dependency set to 0.9.2 instead of the current pre-release version.

pjfanning commented on src/test/scala/sorm/test/general/FieldTestSuite1.scala:
```diff
@@ -32,7 +32,7 @@ class FieldTestSuite1 extends FunSuite with ShouldMatchers with SequentialNested
   test("H2 initialization") { instance(DbType.H2).close() }
   test("MySQL initialization") { instance(DbType.Mysql).close() }
-  test("Oracle initialization") { instance(DbType.Oracle).close() }
+  //test("Oracle initialization") { instance(DbType.Oracle).close() }
```
Is there a reason to turn this part of test off?

takezoux2 replied:
Oracle tests ware broken, and I can't fix it because i'm not familiar with Oracle database.

pjfanning commented on src/main/scala-2.10/sorm/macroimpl/QuerierMacro.scala:
```diff
@@ -0,0 +1,82 @@
+package sorm.macroimpl
```
If we don't continue to do sorm 2.10 builds, do we need this class?

pjfanning commented:
@takezoux2 I did some testing of sorm 0.3.21 and found it only works in scala 2.11. It does look like for 2.12 support, we will need the cross compile support. What do you think it will take to get the sbt stuff merged?

pjfanning commented:
I have pushed a sorm_2.12 to the sonatype snapshots repo, based on https://github.com/scala212-forks/sorm.

0x6C38 commented:
Any reason this hasn't been merged to master?
